src/tree-component.vue

```diff
@@ -20,13 +21,15 @@
                 @drop.prevent.stop="onDropNode(treeViewItem, $event, onDropValidator, treeState)"
                 @dragover.stop="addHoverClass"
                 :checkboxStyle="checkboxStyle"
-                :isCheckable="isCheckable"
+                :isCheckable="isCheckable || treeViewItem.checkable"
```

Comment:
Previously one could either select (one) or check items.

With this change, the user can both select and check at the same time.

Items within the tree that have `checkable` set to `false` will operate with selection. And items with checkable will operate by check?

Reply:
I think it makes sense to support both independently. It looks like it already tracks 'checked' and 'selected', and the tree will emit checkedItems and selectedItems. I didn't realize that making it checkable would disable selection, so I can change that unless you'd rather keep it where it can be selected or checkable, but not both. Just let me know and I can update

Reply:
I think having the `checkable` as a property of the node means

 A child might be checked, and the parent has no hint that one of its children is checked (because it is not checkable). With `checkable` on the tree level, a parent can hint that a child or all its children are checked. 
 
A parent cannot simply be checked to check all it's sub-nodes. They all have to be checkable. 

Let's avoid that complication and simplify, A user can either select (one at a time) or check (zero or many). This includes standard UX patterns like checking a parent means checking all children and vice versa.